sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -324,6 +328,41 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
     }
   }
 
+  // Wait for command spawned via Process
+  private def waitForProcess(cmd: Process, durationMs: Long): Option[Int] = {
+    val endTime = System.currentTimeMillis() + durationMs
+    do {
+      Thread.sleep(10)
+      if (!cmd.isAlive()) {
+        return Some(cmd.exitValue())
+      }
+    } while (System.currentTimeMillis() < endTime)
+    None // Timed out
+  }
+
+  // Try to run nvidia-smi when task fails due to a cuda exception.
+  private def logGpuDebugInfoAndExit(systemExitCode: Int) = synchronized {
+    try {
+      val nvidiaSmiStdout = new StringBuilder
+      val nvidiaSmiStderr = new StringBuilder
+      val cmd = "nvidia-smi".run(
+        ProcessLogger(s => nvidiaSmiStdout.append(s + "\n"), s => nvidiaSmiStderr.append(s + "\n")))
+      waitForProcess(cmd, 2000) match {
```

Comment:
should this be something like 10 seconds? Especially during process launches, nvidia-smi can take a long time to start up.

Reply:
I can definitely make it longer.  Since we are exiting, I think it is safe to do so.